sdk/dapp-kit/test/mockWallet.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Ed25519Keypair } from '@mysten/sui.js/keypairs/ed25519';
+import type {
+	StandardConnectFeature,
+	StandardConnectMethod,
+	StandardDisconnectFeature,
+	StandardDisconnectMethod,
+	StandardEventsFeature,
+	StandardEventsOnMethod,
+	SuiFeatures,
+	SuiSignAndExecuteTransactionBlockMethod,
+	SuiSignPersonalMessageMethod,
+	SuiSignTransactionBlockMethod,
+} from '@mysten/wallet-standard';
+import { ReadonlyWalletAccount, SUI_CHAINS } from '@mysten/wallet-standard';
+import type { Wallet } from '@mysten/wallet-standard';
+
+export class MockWallet implements Wallet {
+	version = '1.0.0' as const;
+	icon = `data:image/png;base64,` as const;
+	chains = SUI_CHAINS;
+	#walletName: string;
+
+	constructor(name: string) {
+		this.#walletName = name;
+	}
+
+	get name() {
+		return this.#walletName;
+	}
+
+	get accounts() {
+		const keypair = new Ed25519Keypair();
+		const account = new ReadonlyWalletAccount({
+			address: keypair.getPublicKey().toSuiAddress(),
+			publicKey: keypair.getPublicKey().toSuiBytes(),
+			chains: ['sui:unknown'],
+			features: ['sui:signAndExecuteTransactionBlock', 'sui:signTransactionBlock'],
+		});
+		return [account];
+	}
+
+	get features(): StandardConnectFeature &
+		StandardDisconnectFeature &
+		StandardEventsFeature &
+		SuiFeatures {
+		return {
+			'standard:connect': {
+				version: '1.0.0',
+				connect: this.#connect,
+			},
+			'standard:disconnect': {
+				version: '1.0.0',
+				disconnect: this.#disconnect,
+			},
+			'standard:events': {
+				version: '1.0.0',
+				on: this.#on,
+			},
+			'sui:signPersonalMessage': {
+				version: '1.0.0',
+				signPersonalMessage: this.#signPersonalMessage,
+			},
+			'sui:signTransactionBlock': {
+				version: '1.0.0',
+				signTransactionBlock: this.#signTransactionBlock,
+			},
+			'sui:signAndExecuteTransactionBlock': {
+				version: '1.0.0',
+				signAndExecuteTransactionBlock: this.#signAndExecuteTransactionBlock,
+			},
+		};
+	}
+
+	#on: StandardEventsOnMethod = () => {
```

Comment:
FYI: All of these mocked functions will probably change once I start getting the PRs up for the related connection/signing hooks. We could run these tests as integration/e2e tests using a fullnode, but I think this is simpler to get going and right now I'm more focused on testing that the the wallet context state is correctly updated.

Reply:
Why not make these spies so that you can mock them out easily?

Reply:
Because ~I'm bad~ wasn't thinking during writing this code, I updated these functions to just be mocks 😅